framework/src/modules/pos/module.ts

```diff
@@ -612,9 +615,9 @@ export class PoSModule extends BaseModule {
 	}
 
 	private async _updateValidators(context: BlockAfterExecuteContext): Promise<void> {
-		const round = new Rounds({ blocksPerRound: this._moduleConfig.roundLength });
 		const { height } = context.header;
-		const nextRound = round.calcRound(height) + 1;
+		const round = await this.method.getRoundNumberFromHeight(context.getMethodContext(), height);
+		const nextRound = round + 1;
```

Comment:
Comment for line 630: In the new PR we realized that `snapshot.validatorWeightSnapshot` does not need to be an argument of `getActiveValidators`, since it can be recovered by the roundNumber and accessing the snapshot. But if for implementation reasons it's more efficient to pass the snapshot as argument (to avoid extra state reads) that's fine for me.

Reply:
I htink it's better to pass it since it will be additional cost

Reply:
resolved